File: 1411-convert-binary-number-in-a-linked-list-to-integer/1411-convert-binary-number-in-a-linked-list-to-integer.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public int getDecimalValue(ListNode head) {
        String res="";
        while(head!=null){
            res+=head.val;
            head=head.next;
        }
        int r=0;
        for(char c:res.toCharArray()){
            r=r*2+(c-'0');
        }
        return r;
    }
}

File: 1411-convert-binary-number-in-a-linked-list-to-integer/1411-convert-binary-number-in-a-linked-list-to-integer.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def getDecimalValue(self, head: Optional[ListNode]) -> int:
        r=""
        curr=head
        while curr:
            r+=str(curr.val)
            curr=curr.next
        res = 0
        n = len(r)
        for i in range(n):
            res += int(r[i]) * (2 ** (n - i - 1)) 
        return res